### PR TITLE
Move app instances into their respective modules

### PR DESCRIPTION
**Pull request by dcramer**
Result of playing around with the idea of killing off sentry.app

The monstrous hack of LazyBackendWrapper has benefits that its much easier to do module level imports (where we can't today due to circular imports).

As a followup we should:

- move sentry.utils.locking into sentry.locks
- rename sentry.ratelimiter to sentry.ratelimits
- create a base Service class which manages enabled() and validate() core methods
- add a basic service registry to aid with things like "validate all services" (this might be able to use the provider code)

#nochanges

### CODE REVIEW

**Comment by dcramer** [comment]
DONT WORRY! Its only like 500 lines changed... across 65 files.

**Review by mitsuhiko — COMMENTED** [review]
Seems good but i want to play around if the expose can be done better.

**Comment by dcramer** [comment]
tests pass in sentry + getsentry, and some minor QA (clicking UI, workers doing work) seems to indicate things are good

File: src/sentry/app.py [hunk 1]
```diff
@@ -25,36 +22,16 @@ class State(local):
 
 env = State()
 
-
-def get_instance(attribute, options, dangerous=()):
-    value = getattr(settings, attribute)
-
-    cls = import_string(value)
```
Comment (dcramer) [inline src/sentry/app.py]: gotta reimplement this real quick

Comment (dcramer) [inline src/sentry/app.py]: resolved

**Comment by dcramer** [comment]
getting a few more quick tests in place then will get this merged

**Review by mattrobenolt — APPROVED** [review]
Nothing at all could go wrong here. :1st_place_medal:

**Pull request merged**